packages/types/src/primitive/Method.ts

```diff
@@ -142,12 +202,32 @@ export default class Method extends Struct implements IMethod {
   //
   // As a convenience helper though, we return the full constructor function,
   // which includes the meta, name, section & actual interface for calling
+  //
+  // @deprecated This method does a lookup in the methods' metadata stored
+  // globally with injectMethods. Pass the metadata to the Method constructor
+  // instead.
   static findFunction (callIndex: Uint8Array): MethodFunction {
     assert(Object.keys(injected).length > 0, 'Calling Method.findFunction before extrinsics have been injected.');
 
     return injected[callIndex.toString()] || FN_UNKNOWN;
   }
 
+  /**
+   * Retrieves a function from the runtime metadata
+   * @param callIndex Call index of the function, e.g. new Uint8Array([3, 0])
+   * @param metadata Runtime metadata, e.g. api.runtimeMetadata
+   */
+  static findByCallIndex (callIndex: Uint8Array, metadata: Metadata): MethodFunction | undefined {
```

Comment:
I think we can remove this. Types should be as simple as possible. If requested by a user, we could add this function as a helper function somewhere else.

Reply:
findByCallIndex is used internally in decodeMethod, to extract the MethodFunction from the runtime metadata + call index

Reply:
Ah yeah, I see. Maybe private static then? Or just a function